File: core-java-practice/leet-code-codebase/reverseString.java
import java.util.*;

 public class reverseString {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);   // Create Scanner object
        String str = sc.nextLine();             // Read input string
        String rev = "";                        // Variable to store reversed string
        // Loop from last character to first
        for (int i = str.length() - 1; i >= 0; i--) {
            rev = rev + str.charAt(i);          // Add characters in reverse order
        }
        System.out.println("Reversed string: " + rev);

    
    }
}

File: dsa-csharp-practice/csharp-linear-and-binary-search/ReverseString.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BridgeLabzTraining.dataStructures.linearBinarySearch
{
    internal class ReverseString
    {
        public static void Main(string[] args)
            {
                // Ask user to enter a string
                Console.Write("Enter a word or sentence: ");
                string input = Console.ReadLine();

                // Create empty StringBuilder for reversed string
                StringBuilder reversed = new StringBuilder();

                // Loop from last character to first
                for (int i = input.Length - 1; i >= 0; i--)
                {
                    reversed.Append(input[i]); // add character one by one
                }

                Console.WriteLine("Reversed string: " + reversed.ToString());
            }
        }
    }




